packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -281,8 +281,18 @@ export default class OptionManager {
           val = require(presetLoc);
         }
 
-        // If the imported preset is a transpiled ES2015 module, grab the default export.
-        if (typeof val === "object" && val.__esModule) val = val.default;
+        // If the imported preset is a transpiled ES2015 module
+        if (typeof val === "object" && val.__esModule) {
+          // Try to grab the default export.
+          if (val.default) {
+            val = val.default;
+          } else {
+            // If there is no default export we treat all named exports as options
```

Comment:
So we support return a default export that is either an object or a function? I add a comment.